[IOPLT-1066] Upgrade packages

robbcocco opened the pull request:
<!--- Please always add a PR description as if nobody knows anything about the context these changes come from. -->
<!--- Even if we are all from our internal team, we may not be on the same page. -->
<!--- Write this PR as you were contributing to a public OSS project, where nobody knows you and you have to earn their trust. -->
<!--- This will improve our projects in the long run! Thanks. -->

#### List of Changes
Upgraded lint config and packages
<!--- Describe your changes in detail -->

#### Motivation and Context
This change was needed to upgrade the stack
<!--- Why is this change required? What problem does it solve? -->

#### How Has This Been Tested?
Tested by running yarn code-review
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

#### Screenshots (if appropriate):

<!--- Attach screenshots in case changes impact UI. -->

#### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

roberto-cocco-ey committed:
Update io-sign package

roberto-cocco-ey committed:
Fix io-sign lint

roberto-cocco-ey committed:
Update backoffice-func package

roberto-cocco-ey committed:
Fix backoffice-func lint

roberto-cocco-ey committed:
Update sign-issuer package

roberto-cocco-ey committed:
Fix sign-issuer lint

roberto-cocco-ey committed:
Update sign-user package

roberto-cocco-ey committed:
Fix sign-user lint

roberto-cocco-ey committed:
Update sign-support package

roberto-cocco-ey committed:
Fix sign-support lint

roberto-cocco-ey committed:
Align and update projects packages

roberto-cocco-ey committed:
Update node version

roberto-cocco-ey committed:
Add changeset

roberto-cocco-ey committed:
Hardlink eslint-config version

lucacavallaro reviewed (dismissed):
Hi @robbcocco,

To make these changes easier to review, I’d like to ask you to separate them into two different PRs: one for the `Node.js` update, and another one for `prettier`, `eslint`, etc.

From a quick glance, I can already point out a few things that should be changed:
1. The latest version of node is `20.19.2`
2. You must not change the `prettier` version or its configuration unless there are compatibility issues, which you must explain and document.
3. We have a shared, battle-tested `eslint` configuration: https://www.npmjs.com/package/@pagopa/eslint-config. It is compatible with `eslint@8`, designed to be "automatic" and mostly "auto-fixable", so there's no need to define custom rules.
4. `type` must not be replaced with `interface` — they behave differently, and using `type` here was intentional.

roberto-cocco-ey committed:
Revert "Fix sign-support lint"

This reverts commit 44f89b7ef5883b80227097689bc03048c5a6e56d.

roberto-cocco-ey committed:
Revert "Fix sign-user lint"

This reverts commit 5b33e72a538e087d406f01522b5b1e9a172a4fbc.

roberto-cocco-ey committed:
Revert "Fix sign-issuer lint"

This reverts commit 2ecf9acf339b0b9d4ec215ef9fdb84e1c588f009.

roberto-cocco-ey committed:
Revert "Fix backoffice-func lint"

This reverts commit eed9b595f54c1f82d3fe379ed0dc2f2030cb0cf7.

roberto-cocco-ey committed:
Revert "Fix io-sign lint"

This reverts commit c5df4408ce9fe1b1b02660e992ee77e00de4dc44.

roberto-cocco-ey committed:
Update node version to 20.19.2

roberto-cocco-ey committed:
Add io-sign/eslint-config package

roberto-cocco-ey committed:
Update eslint config package.json
Update esling config rules

roberto-cocco-ey committed:
Fix backoffice-func lint

roberto-cocco-ey committed:
Fix sign-issuer lint

roberto-cocco-ey committed:
Fix sign-support lint

roberto-cocco-ey committed:
Fix sign-user lint

roberto-cocco-ey committed:
Fix io-sign/io-sign lint

roberto-cocco-ey committed:
Lock eslint-config version

robbcocco commented:
> Hi @robbcocco,
> 
> To make these changes easier to review, I’d like to ask you to separate them into two different PRs: one for the `Node.js` update, and another one for `prettier`, `eslint`, etc.

> **I'll make a separate PR for the changes regarding only the Node.js version.**

> 
> From a quick glance, I can already point out a few things that should be changed:
> 
> 1. The latest version of node is `20.19.2`

> **I've upgraded the requirements and the version**

> 2. You must not change the `prettier` version or its configuration unless there are compatibility issues, which you must explain and document.

> **Upgrading to the latest `@pagopa/eslint-config` requires prettier 3.x. The change is not necessary in itself (upgrading nodejs only should not break compatibility) but I thought it was a nice to have among others libraries upgrades**

> 3. We have a shared, battle-tested `eslint` configuration: https://www.npmjs.com/package/@pagopa/eslint-config. It is compatible with `eslint@8`, designed to be "automatic" and mostly "auto-fixable", so there's no need to define custom rules

> **The latest version (4.x) introduces new rules (and likely replaces or removes others), so with @BurnedMarshall we came up with a set of custom rules to minimize the changes migrating from version 3.x to 4.x. To ease the transition in this monorepo, we also decided to make an internal package from which the projects will inherit the configuration**

> 4. `type` must not be replaced with `interface` — they behave differently, and using `type` here was intentional.

> **Continuing on point 3, I've added a rule to block this "fix" from the linter and reverted the changes**

Hi @lucacavallaro, thanks for the feedback.
I've yet to fix some of the issues you reported, in the meantime I've replied inline with my considerations on some of the biggest changes and other issues i've tackled.

roberto-cocco-ey committed:
Revert "Update node version"

This reverts commit 2da2d5e8c76214891adc9c1481e06020c9853a71.

roberto-cocco-ey committed:
Update changeset

BurnedMarshal commented on apps/io-func-sign-issuer/package.json:
```diff
@@ -25,6 +25,7 @@
   "dependencies": {
     "@azure/cosmos": "^4.0.0",
     "@azure/event-hubs": "^5.9.0",
+    "@azure/functions": "^3.5.0",
```
Double check why this dependency is moved from `devDependencies` in here. If only types are used it should remain as devDependency. There are other cases because you promote it as dependency?

robbcocco replied:
Moved back to devDependencies in ff08dcdc4b3432549c28ff9ca32f68d2ed2da4e5

BurnedMarshal commented on apps/io-func-sign-issuer/src/infra/back-office/issuer.ts:
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
```
```suggestion
```
Disable this rule locally, not globally for the file.

BurnedMarshal reviewed (changes requested):
Only a couple of comments.

roberto-cocco-ey committed:
Add comment for file lint disable
Moved @azure/functions back to devDependencies in sign-issuer

robbcocco committed:
Merge branch 'main' into IOPLT-1066_upgrade_to_node_20

roberto-cocco-ey committed:
Remove unnecessary dependencies

roberto-cocco-ey committed:
Merge branch 'main' into IOPLT-1066_upgrade_to_node_20

roberto-cocco-ey committed:
Update @pagopa/eslint-config library

BurnedMarshal committed:
Merge branch 'main' into IOPLT-1066_upgrade_to_node_20

robbcocco committed:
Merge branch 'main' into IOPLT-1066_upgrade_to_node_20

BurnedMarshal reviewed (approved):
LGTM
